### ISSUE TITLE
In order to [objectify courriers waiting time], I want to [add a "WAITING" button in the application].

### ISSUE THREAD

**Issue by rayon9coop** [coopcycle/coopcycle-web#3455]
**To objectify the time spent waiting for a parcel. 
Sometimes things don't work as they should...
The courier is waiting for the parcel because the customer says "ok, the parcel will be ready in two minutes" but those two minutes don't seem to be the same as the ones on the courier's watch :-)

**As a** `role
As a dispatch or an administrative's member of a coop, if I have access to the waiting time, I can present it to my customer, change something in the process to avoid delays or possibly introduce compensation if delays are systematives.

**I want to** "add or change something".
Adding a "waiting" button in the application could be useful. It could be added on the task screen, right next to the time slot (see image below). When the courier starts waiting, he presses the button and the waiting time starts to run. The waiting time is stopped by pressing the button again or by the end of the job (which means that the courier is no longer waiting). The wait time should be displayed on the POD screen (to certify that the person signing the app has seen it) and should be added to the data in the task export for invoice's work.

![Screenshot_20221024-123647_CoopCycle jpg](https://user-images.githubusercontent.com/44605344/197511697-9d8aac38-b30f-4586-88b3-52b6075d8bba.png)

**Comment by biologeek** [comment]
Hi, maybe using the "start" task command at the beginning of waiting and "complete" when delivered would fit your need ?

![Screenshot_20221024152314](https://user-images.githubusercontent.com/11749973/197536392-a5318a83-ea23-4a93-9244-598442e241b8.png)

Thus logging task start and end in database could do the job @alexsegura ?

**Comment by ERAMANrepartos** [comment]
This is a workaround. Not a bad one really, but still a workaround. 

In food delivery, we are currently commenting "Listo" [ready] or "Espera" [wait] when doing pickups, which serves a similar purpose. 

The real solution I see would be a feature in which for food delivery or manual deliveries the provider can push a button that says "ready" and shows that a delivery is now prepared. This change in status would be reflected both in the orders panel and the admin dashboard visually so that a dispatcher could see this change visually.

**Comment by rayon9coop** [comment]
Why not but this will not avoid waiting time because people are humain and make mistakes (= thinking a order is ready and in fact missing a article). Objectify waiting time could be helpfull whatever the mean used.

**Comment by ERAMANrepartos** [comment]
Good point. Maybe check out the discussion about incidents and reporting? I think this could be solved by some of the suggestions there

**Comment by Paul-Eraman-CoopCycle** [comment]
@alexsegura can we add a "task started at" column in the universal excel?
